pkg/config/config.go

```diff
@@ -57,43 +67,27 @@ func NewKVConfiguration(fileName string) (*KVConfiguration, error) {
 	return configuration, err
 }
 
-func (c *KVConfiguration) GetReplications() []string {
-	var localStores []string
-	var neighborStores []string
-	var remoteStores []string
-	// Please refer to https://user-images.githubusercontent.com/252020/184443299-799f1ed4-493a-4ea2-a9ed-72e15a2737af.png
-	// for the following store categories.
+// Returns local stores grouping by AvailabilityZone and remote stores in array
+func (c *KVConfiguration) GetReplications(remoteStoreLatencyThreshold int64) (map[constants.AvailabilityZone][]string, []string, error) {
+	localStores := make(map[constants.AvailabilityZone][]string)
+	remoteStores := make([]string, 0)
 	for _, store := range c.Stores {
-		switch region := store.RegionType; region {
-		case "local":
-			localStores = append(localStores, store.Name)
-		case "neighbor":
-			neighborStores = append(neighborStores, store.Name)
-		case "remote":
-			remoteStores = append(remoteStores, store.Name)
-		}
-	}
-	n := len(localStores)
-	replications := make([]string, n)
-	for idx := 0; idx < n; idx++ {
-		total := c.ReplicaNum
-		replics := make([]string, total)
-		total--
-		if remote := selectRandom(remoteStores); remote != "" {
-			replics[total] = remote
-			total--
-		}
-		if neighbor := selectRandom(neighborStores); neighbor != "" {
-			replics[total] = neighbor
-			total--
-		}
-		for i := 0; i < total+1; i++ {
-			replics[total] = localStores[(idx+i)%n]
-			total--
+		target := fmt.Sprintf("%s:%d", store.Host, store.Port)
+		if latency, err := latency.GetLatency(target, 10); err != nil {
+			return localStores, remoteStores, fmt.Errorf("failed to get latency from %s", target)
+		} else {
```

Comment:
this "else" not needed as there's a "return" on line 77

Reply:
Updated.